script/testing/junit/src/TracefileTest.java

```diff
@@ -118,6 +133,43 @@ public static DynamicTest executeSelectQuery(int num, String cur_sql) {
         return cur;
     }
 
+    public static void checkEquals(List<String> res, List<String> queryResult) throws Exception {
+        if(res.size()!=queryResult.size()){
+            throw new Exception("Query got wrong number of results");
+        }
+        double precision = 0.000001;
+        for(int i=0;i<res.size();i++){
+            if(res.get(i)==null){
+                if(queryResult.get(i)!=null){
+                    throw new Exception("Value null Mismatch");
+                }
+            }else if(res.get(i).equals("")){
+                if(!queryResult.get(i).equals("")){
+                    throw new Exception("Value '' Mismatch");
+                }
+            }else{
+                try{
+                    double one = Double.parseDouble(res.get(i));
+                    double two = Double.parseDouble(queryResult.get(i));
+                    if(Math.abs(one-two)>precision){
+                        throw new Exception("Expected " + queryResult + " but have " + res);
+                    }
```

Comment:
As we talked about, we don't want to do this rounding. We want to match our own output.

Reply:
Yeah there's no rounding here. Within this function I compare the query result list to the list from trace.

Reply:
What if the value is a string and not a double?

Reply:
Then it will enter the catch clause below comparing the values.